javascript/modules/client/src/client.ts

```diff
@@ -579,50 +593,152 @@ export class Client extends ClientCore implements IClient {
+      });
+      for (let i = 0; i < daos.length; i++) {
+        const dao = daos[i];
+        // const stringMetadata = await this.ipfs.fetchString(dao.metadata);
+        // TODO
+        // this is a temporal fix and should be changed by the line above
+        // but the current daos in subgraph dont have a proper metadata
+        const stringMetadata = await this.ipfs.fetchString(
+          "QmebY8BGzWBUyVqZFMaFkFmz3JrfaDoFP5orDqzJ1uiEkr",
+        );
+        const metadata = JSON.parse(stringMetadata);
+        daosDetails.push({
+          address: dao.id,
+          ensDomain: dao.name,
+          metadata: {
+            name: metadata.name,
+            avatar: metadata.avatar || null,
+          },
+          // TODO update when new subgraph schema is deployed
+          plugins: dao.packages.map(
+            (
+              pkg: { pkg: { id: string; __typename: string } },
+            ): InstalledPluginListItem => {
+              return {
+                instanceAddress: pkg.pkg.id,
+                id: pkg.pkg.__typename,
+                version: "0.0.1",
+              };
+            },
+          ),
+        });
+      }
+      return daosDetails;
+    } catch (err) {
+      throw new Error("Cannot fetch the dao details data from GraphQL");
     }
-    return new Promise((resolve) => setTimeout(resolve, 1000)).then(
-      () => (metadataMany),
-    );
   }
 
-  private _getBalances(
-    daoIdentifier: string,
+  private async _getBalances(
+    daoAddress: string,
     _tokenAddresses: string[],
-  ): Promise<AssetBalance[]> {
-    // TODO: Implement actual fetch logic using subgraph.
-    // Note: it would be nice if the client could be instantiated with dao identifier
-
-    if (!daoIdentifier) {
-      throw new Error("Invalid DAO address or ENS");
+  ): Promise<AssetBalance[] | null> {
+    // TODO check address
+    try {
+      await this.graphql.ensureOnline();
+      const client = this.graphql.getClient();
+      const { dao } = await client.request(QueryBalances, {
+        address: daoAddress,
+      });
+      if (!dao) {
+        return null;
+      }
+      if (dao.balances.length === 0) {
+        return [];
+      }
+      const balances: AssetBalance[] = [];
+      for (let i = 0; i < dao.balances.length; i++) {
+        const balance = dao.balances[i];
+        if (balance.token.symbol === "ETH") {
+          balances.push({
+            type: "native",
+            balance: BigInt(balance.balance),
+            lastUpdate: new Date(parseInt(balance.lastUpdated) * 1000),
+          });
+          continue;
+        }
+        balances.push({
+          type: "erc20",
+          address: balance.token.id,
+          name: balance.token.name,
+          symbol: balance.token.symbol,
+          decimals: parseInt(balance.token.decimals),
+          balance: BigInt(balance.balance),
+          lastUpdate: new Date(parseInt(balance.lastUpdated) * 1000),
+        });
+      }
+      return balances;
```

Comment:
If we're not handling the extra tokens here, do we have a task to address that in subgraph and then the SDK? Yet, a TODO comment here?